src/main/java/com/internship/backend/service/UserService.java

```diff
@@ -4,119 +4,95 @@
 import com.internship.backend.exceptions.UserAlreadyExistsException;
 import com.internship.backend.exceptions.UserDoesNotExistException;
 import com.internship.backend.model.Authority;
-import com.internship.backend.model.Users;
+import com.internship.backend.model.User;
 import com.internship.backend.repository.AuthorityRepository;
 import com.internship.backend.repository.UserRepository;
-import jakarta.transaction.Transactional;
-import lombok.RequiredArgsConstructor;
-import org.apache.commons.lang3.ObjectUtils;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.security.core.userdetails.MapReactiveUserDetailsService;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
-import java.util.HashSet;
-import java.util.Iterator;
 import java.util.List;
 import java.util.Set;
-
+import java.util.UUID;
 
 @Service
-@RequiredArgsConstructor
 public class UserService {
 
     @Autowired
     private UserRepository userRepository;
 
     @Autowired
-    private PasswordEncoder passwordEncoder;
-
+    private AuthorityRepository authorityRepository;
 
     @Autowired
-    private AuthorityRepository authorityRepository;
-    private MapReactiveUserDetailsService reactiveUserDetailsService;
+    private PasswordEncoder passwordEncoder;
 
-    @Transactional
-    public List<Users> getAllUsersProcedure(){
-        return userRepository.getAllUsersProcedure();
+    public User register(UserDTO userDTO) throws UserAlreadyExistsException {
+        return register(addIdUsernamePasswordEmailAuthorityToUser(userDTO));
     }
 
-    @Transactional
-    public void deleteUserByIdProcedure(Integer userId){
-        authorityRepository.deleteByUserId(userId);
-        userRepository.deleteUserByIdProcedure(userId);
+    public User register(User user) {
+        return userRepository.save(user);
     }
 
-    public Users register(Users user) throws UserAlreadyExistsException {
-        if (userRepository.findByUsername(user.getUsername()) != null) {
+    private User addIdUsernamePasswordEmailAuthorityToUser(UserDTO userDTO) throws UserAlreadyExistsException {
```

Comment:
you could rewrite this method : 
private User addIdUsernamePasswordEmailAuthorityToUser(UserDTO userDTO) throws UserAlreadyExistsException {
        checkUserExists(userDTO);

        var user = createUser(userDTO);

        user.setAuthorities(of(createAuthority(userDTO)));

        authorityRepository.save(authority);

        return user;
    }
and to implement checkUserExists, createUser, createAuthority..